src/editor/Editor.jsx

```diff
@@ -6,49 +6,195 @@ var TextFieldRich = require("../TextFieldRich.jsx");
+                    			<FontIcon onClick={this._displayColorPicker.bind(this, false, true)} className="cfi cfi-times" />
+                    		</div>
+                    		<div>
+                    			<ColorPicker onChange={this._onColorPick} />
+                    		</div>
+                    	</div>
+                	</ToolbarGroup>
+                    <ToolbarGroup key={5} float="right">
+                        <DropDownIcon 
+                        	iconClassName="cfi cfi-header" 
+                        	menuItems={fontStyleOptions}
+                        	onChange={this._onMenuClick.bind(this, "formatblock")} />
+                        <DropDownIcon 
+	                    	iconClassName="cfi cfi-text-height" 
+	                    	menuItems={fontSizeOptions}
+	                    	onChange={this._onMenuClick.bind(this, "fontsize")} />
+                        <DropDownIcon 
+                        	iconClassName="cfi cfi-font" 
+                        	menuItems={fontNameOptions}
+                    		onChange={this._onMenuClick.bind(this, "fontname")} />
+                </ToolbarGroup>
                 </Toolbar>
                 <div>
-                    <TextFieldRich />
+                    <TextFieldRich ref="rte" />
                 </div>
+                <Dialog ref="linkDialog" title="Enter the link path" actions={dialogActions} modal={true} >
+                	<TextField ref="linkInput" />
+                </Dialog>
             </div>
         )
     },
 
+    /**
+     * Callback used to handle commands when button is clicked on the toolbar
+     *
+     * @param {string} type		The name of the command to execute
+     * @private
+     */
+    _onToolbarClick: function(type) {
+    	if (!this.isMounted()) { 
+    		return;
+    	}
+    	
+    	switch(type) {
+    		case "src":
+    			this.refs.rte._toggleSrc();
+    			break;
+    		case "link":
+    			this.refs.linkDialog.show();
+    			break;
+    		case "table":
+    			this.refs.rte._insertHtml('<table style="border: 1px solid; padding: 10px;"><tbody><tr><td>&nbsp;</td><td>&nbsp;</td></tr></tbody></table>');
+    			break;
+    		case "backcolor":
+    		case "forecolor":
+    			this._displayColorPicker(this._colorFontType == type, false);
+    			this._colorFontType = type;
+    			break;
+    		default: 
+    			this.refs.rte._sendCommand(type, '');
+    			break;
+    	}
+    },
+    
+    /**
+     * Callback used to handle commands when button is clicked on the toolbar
+     *
+     * @param {string} type		The name of the command that triggered the menu click
+     * @param {DOMEvent} e 		Reference to the DOM event being sent
+     * @param {Integer} key		The index of the menu clicked
+     * @param {Object} payload	The object value of the menu clicked
+     * @private
+     */
+    _onMenuClick: function(type, e, key, payload) {
+    	this.refs.rte._sendCommand(type, payload.payload);
```

Comment:
We should never call a private function of another class. We'll need to make sendCommand public.